llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -797,6 +797,16 @@ SDValue TargetLowering::SimplifyMultipleUseDemandedBits(
       return Op.getOperand(1);
     break;
   }
+  case ISD::ADD: {
```

Comment:
Should this also cover or and sub?

Reply:
OR is already handled more generally - I'll see if I can get (SUB X, 0) to match in any tests